Sources/ComposableArchitecture/SwiftUI/TextState.swift

```diff
@@ -0,0 +1,304 @@
+    var `self` = self
+    `self`.modifiers.append(.baselineOffset(baselineOffset))
+    return `self`
+  }
+
+  public func bold() -> Self {
+    var `self` = self
+    `self`.modifiers.append(.bold)
+    return `self`
+  }
+
+  public func font(_ font: Font?) -> Self {
+    var `self` = self
+    `self`.modifiers.append(.font(font))
+    return `self`
+  }
+
+  public func fontWeight(_ weight: Font.Weight?) -> Self {
+    var `self` = self
+    `self`.modifiers.append(.fontWeight(weight))
+    return `self`
+  }
+
+  public func foregroundColor(_ color: Color?) -> Self {
+    var `self` = self
+    `self`.modifiers.append(.foregroundColor(color))
+    return `self`
+  }
+
+  public func italic() -> Self {
+    var `self` = self
+    `self`.modifiers.append(.italic)
+    return `self`
+  }
+
+  public func kerning(_ kerning: CGFloat) -> Self {
+    var `self` = self
+    `self`.modifiers.append(.kerning(kerning))
+    return `self`
+  }
+
+  public func strikethrough(_ active: Bool = true, color: Color? = nil) -> Self {
+    var `self` = self
+    `self`.modifiers.append(.strikethrough(active: active, color: color))
+    return `self`
+  }
+
+  public func tracking(_ tracking: CGFloat) -> Self {
+    var `self` = self
+    `self`.modifiers.append(.tracking(tracking))
+    return `self`
+  }
+
+  public func underline(_ active: Bool = true, color: Color? = nil) -> Self {
+    var `self` = self
+    `self`.modifiers.append(.underline(active: active, color: color))
+    return `self`
+  }
+}
+
+extension Text {
+  public init(_ state: TextState) {
+    let text: Text
+    switch state.storage {
+    case let .concatenated(first, second):
+      text = Text(first) + Text(second)
+    case let .localized(content, tableName, bundle, comment):
+      text = .init(content, tableName: tableName, bundle: bundle, comment: comment)
+    case let .verbatim(content):
+      text = .init(verbatim: content)
+    }
+    self = state.modifiers.reduce(text) { text, modifier in
+      switch modifier {
+      case let .baselineOffset(baselineOffset):
+        return text.baselineOffset(baselineOffset)
+      case .bold:
+        return text.bold()
+      case let .font(font):
+        return text.font(font)
+      case let .fontWeight(weight):
+        return text.fontWeight(weight)
+      case let .foregroundColor(color):
+        return text.foregroundColor(color)
+      case .italic:
+        return text.italic()
+      case let .kerning(kerning):
+        return text.kerning(kerning)
+      case let .strikethrough(active, color):
+        return text.strikethrough(active, color: color)
+      case let .tracking(tracking):
+        return text.tracking(tracking)
+      case let .underline(active, color):
+        return text.underline(active, color: color)
+      }
+    }
+  }
+}
+
+extension TextState: View {
+  public var body: some View {
+    Text(self)
+  }
+}
+
+extension String {
+  public init(state: TextState, locale: Locale? = nil) {
```

Comment:
Added this just to keep our UIKit compatibility layer. It loses formatting. Dunno if it's possible to recover with attributed strings, but there are certain SwiftUI constructs that don't translate over (while `Color` can convert to `UIColor`, `Font` does not seem to convert to `UIFont`, etc.).

Reply:
yeah seems ok. eventually we should maybe create UIKit helpers like we have for `.alert(Store)`

Reply:
I did write above mentioned helpers for UIKit, unfortunately this PR has made this extension impossible, due to Button's `type` becoming internal... Was that strictly necessary?

``` swift
extension AlertState.Button {
  func toUIAlertAction(send: @escaping (Action) -> Void) -> UIAlertAction {
    let action = { if let action = self.action { send(action) } }
    switch self.type {
    case let .cancel(.some(title)):
      return UIAlertAction(title: title.formatted(), style: .cancel, handler: { _ in action() })
    case .cancel(.none):
      return UIAlertAction(title: nil, style: .cancel, handler: { _ in action() })
    case let .default(title):
      return UIAlertAction(title: title.formatted(), style: .default, handler: { _ in action() })
    case let .destructive(title):
      return UIAlertAction(
        title: title.formatted(), style: .destructive, handler: { _ in action() })
    }
  }
}
```

Reply:
And I know that `formatted()` is no more, just pasted the snippet as is, before updating/deleting it due to 0.11.0

Reply:
> Was that strictly necessary?

Ah, definitely not. I had made the (poor) assumption that this API didn't need to be public, but now see how it can be depended upon. I'll revert this part of the change.

Reply:
Thanks!